tools/server/server.cpp

```diff
@@ -3699,6 +3777,13 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
+    PidFile f;
+    if (!params.pidfile.empty()) {
```

Comment:
the variable is not used outside its scope, so it's better to move inside the `if {..}`

```suggestion
    if (!params.pidfile.empty()) {
        PidFile f;
```

Reply:
No there's a reason it's in the outer scope, we want the destructor to attempt to delete the file when main() exits